File: 20242Q/poo/laboratorio/tp04/ej5/src/HTMLTester.java

public class HTMLTester {

    public static void main(String[] args) {
        PlainText text = new PlainText("Hola");
        HTMLText boldText = new BoldText(text);
        HTMLText italicText = new ItalicText(text);
        System.out.println(boldText);
        System.out.println(italicText);
        HTMLText boldItalicText = new BoldText(italicText);
        System.out.println(boldItalicText);
        text.setText("ITBA");
        System.out.println(boldText);
        System.out.println(italicText);
        System.out.println(boldItalicText);
        HTMLText linkText = new LinkText(text, "itba.edu.ar");
        HTMLText linkBoldText1 = new LinkText(boldItalicText, "itba.edu.ar");
        HTMLText linkBoldText2 = new BoldText(linkText);
        System.out.println(linkText);
        System.out.println(linkBoldText1);
        System.out.println(linkBoldText2);
        text.setText("Ejemplo");
        System.out.println(linkBoldText1);
        System.out.println(linkBoldText2);
    }

}


// !!! buen ej
// hacer seguimiento de cómo se utiliza la interfaz HTMLText y cómo se guarda HTMLText como elemento en común

// !!
// es como una relación de recurrencia
// en donde el caso base es que se guarde PlainText en HTMLText element

// obs: para LinkText tengo que hacer un source() aparte, pues es exclusivo

File: 20242Q/poo/laboratorio/tp10/_fuentes/ej1/html_tester.rb
text = PlainText.new 'Hola'
# bold_text = ...
# italic_text = ...
puts bold_text # <b>Hola</b>
puts italic_text # <i>Hola</i>
# bold_italic_text = ...
puts bold_italic_text # <b><i>Hola</i></b>
text.content = 'ITBA'
puts bold_text # <b>ITBA</b>
puts italic_text # <i>ITBA</i>
puts bold_italic_text # <b><i>ITBA</i></b>
# link_text = ...
# link_bold_italic_text = ...
# bold_link_text = ...
puts link_text # <a href:"www.itba.edu.ar">ITBA</a>
puts link_bold_italic_text # <a href:"www.itba.edu.ar"><b><i>ITBA</i></b></a>
puts bold_link_text # <b><a href:"www.itba.edu.ar">ITBA</a></b>
text.content = 'Ejemplo'
puts link_bold_italic_text # <a href:"www.itba.edu.ar"><b><i>Ejemplo</i></b></a>
puts bold_link_text # <b><a href:"www.itba.edu.ar">Ejemplo</a></b>
